### PR TITLE
Gives user visibility into whether nodes are external inputs or not

### PR DESCRIPTION
**Pull request by elijahbenizzy**
This allows you to programmatically determine the inputs to your DAG. One component of #97.

[Short description explaining the high-level reason for the pull request]

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: hamilton/driver.py [hunk 1]
```diff
@@ -69,7 +69,8 @@ class Variable:
 
     name: str
     type: typing.Type
-    tags: Dict[str, str] = field(default_factory=frozenset)
+    tags: Dict[str, str] = field(default_factory=dict)
+    is_external_input: bool = field(default=False)
```
Comment (skrawcz) [inline hamilton/driver.py:73]: we use `UD` -- so should it be `is_user_defined` instead?

Comment (elijahbenizzy) [reply hamilton/driver.py:73]: If anything we should change that -- `user_defined` to me makes less sense than `external`. Also `UD` is confusing without a legend...

Comment (skrawcz) [reply hamilton/driver.py:73]: but maybe `is_external_input` is fine 🤔

Comment (skrawcz) [reply hamilton/driver.py:73]: Yep sure. I'm just thinking about language. @gravesee do you have any preference here?

**Review by skrawcz — APPROVED** [review]

**Pull request merged**